Internal(Components): Modal is not shown when user tries to edit a locked component [ED-21738]

MuliElementor opened the pull request:
- Updated `unlock` method in `Document_Lock_Manager` to include error handling and ensure proper removal of WordPress post locks.
- Improved `is_locked` method to check both custom and WordPress locks, updating lock metadata accordingly.
- Refactored `extend_lock` method to accept optional lock data, reducing redundant calls.
- Enhanced `Component_Lock_Manager` to allow unlocking by users who own the WordPress lock.
- Updated unit tests to reflect changes in locking behavior and ensure correct functionality when locks expire.

## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

<!--start_gitstream_placeholder-->
### ✨ PR Description
Purpose: Fix component locking mechanism to properly display warning modal when users attempt to edit locked components.
Main changes:
- Refactored lock management APIs to use consistent data structure with locked_by/locked_at fields
- Improved expired lock detection with separate is_lock_expired method
- Fixed get_lock_status REST endpoint to correctly identify lock ownership and permissions

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>
<!--end_gitstream_placeholder-->


[ED-21738]: https://elementor.atlassian.net/browse/ED-21738?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

MuliElementor committed:
Enhance document locking mechanism and error handling

- Updated `unlock` method in `Document_Lock_Manager` to include error handling and ensure proper removal of WordPress post locks.
- Improved `is_locked` method to check both custom and WordPress locks, updating lock metadata accordingly.
- Refactored `extend_lock` method to accept optional lock data, reducing redundant calls.
- Enhanced `Component_Lock_Manager` to allow unlocking by users who own the WordPress lock.
- Updated unit tests to reflect changes in locking behavior and ensure correct functionality when locks expire.

elementorbot committed:
Merge branch 'main' into fix-lock

MuliElementor committed:
lint

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

elementorbot committed:
Merge branch 'main' into fix-lock

MuliElementor committed:
update

MuliElementor committed:
Merge branch 'main' into fix-lock

ronros-elementor committed:
Merge branch 'main' into fix-lock

ronros-elementor commented on modules/components/document-lock-manager.php:
```diff
@@ -87,21 +87,34 @@ public function lock( $document_id ) {
-		delete_post_meta( $document_id, self::LOCK_EDIT_LOCK_META );
+			if ( function_exists( 'wp_clear_post_lock' ) ) {
+				call_user_func( 'wp_clear_post_lock', $document_id );
+			} else {
```
did you check that it's even a thing? we never used in in our codebase, and I couldn't find any wp (or any other) documentation of this function - `wp_clear_post_lock`...

MuliElementor committed:
Merge branch 'main' into fix-lock

MuliElementor committed:
Refactor component locking mechanism to improve clarity and functionality

- Updated `Component_Lock_Manager` to use `lock_component` and `unlock_component` methods for better readability.
- Replaced the `is_locked` method with `get_lock_data` to provide clearer lock status information.
- Simplified locking and unlocking logic, ensuring proper checks for component types.
- Enhanced unit tests to reflect changes in method names and functionality.
- Removed deprecated methods and comments for cleaner code.

MuliElementor committed:
updaate

MuliElementor committed:
update tests

MuliElementor committed:
Merge branch 'main' into fix-lock

MuliElementor committed:
lint

MuliElementor committed:
Merge branch 'fix-lock' of https://github.com/elementor/elementor into fix-lock

MuliElementor committed:
lint

MuliElementor committed:
Remove outdated TODO comment in Component_Lock_Manager for improved code clarity

MuliElementor committed:
Merge branch 'main' into fix-lock

MuliElementor committed:
Enhance documentation in Component_Lock_Manager by adding exception details for locking and unlocking methods. This improves clarity on error handling when the post is not a component type.

MuliElementor committed:
Merge branch 'fix-lock' of https://github.com/elementor/elementor into fix-lock

MuliElementor committed:
Merge branch 'main' into fix-lock

MuliElementor committed:
Merge branch 'main' into fix-lock

MuliElementor committed:
Refactor: Update locking mechanism in Component_Lock_Manager and related classes

- Changed ONE_HOUR constant to represent the correct duration in seconds.
- Renamed unlock_component and lock_component methods to unlock and lock for consistency.
- Added checks for lock expiration in lock and unlock methods to ensure proper handling of expired locks.
- Updated Components_REST_API to reflect method name changes and handle lock expiration logic.
- Enhanced Document_Lock_Manager with is_lock_expired method for better lock management.

MuliElementor committed:
Refactor: Standardize lock and unlock method names in Component_Lock_Manager

- Updated method calls from lock_component and unlock_component to lock and unlock for consistency.
- Adjusted the unlock logic to call the parent method for handling expired locks.
- Enhanced unit tests to reflect method name changes and added new tests for lock expiration scenarios.

MuliElementor committed:
fix

MuliElementor commented on modules/components/document-lock-manager.php:
```diff
@@ -98,28 +102,39 @@ public function get_lock_data( $document_id ) {
 	 * @return bool True if extended successfully, false if not locked or locked by another user
 	 */
 	public function extend_lock( $document_id ) {
+		if ( $this->is_lock_expired( $document_id ) ) {
```
removed

MuliElementor committed:
Refactor: Remove expired lock check in Component_Lock_Manager

- Eliminated the check for expired locks in the lock acquisition process to streamline functionality.
- This change simplifies the locking logic while maintaining the integrity of the lock management system.

MuliElementor committed:
fix tests

MuliElementor committed:
Merge branch 'main' into fix-lock

MuliElementor committed:
Merge branch 'main' into fix-lock

MuliElementor committed:
Merge branch 'main' into fix-lock
